models/CMakeLists.txt

```diff
@@ -1,19 +1,30 @@
-# List the subdirectory (http://stackoverflow.com/questions/7787823/cmake-how-to-get-the-name-of-all-subdirectories-of-a-directory)
-macro(SUBDIRLIST result curdir)
-  file(GLOB children RELATIVE ${curdir} ${curdir}/*)
-  set(dirlist "")
-  foreach(child ${children})
-    if(IS_DIRECTORY ${curdir}/${child})
-      list(APPEND dirlist ${child})
-    endif()
-  endforeach()
-  set(${result} ${dirlist})
-endmacro()
-
-# Get list of models 
-subdirlist(subdirs ${CMAKE_CURRENT_SOURCE_DIR})
-
-# Install each model 
-foreach (dir ${subdirs})
-  install (DIRECTORY ${dir} DESTINATION share/gazebo/models)
-endforeach ()
\ No newline at end of file
+# configure_files for all files in a folder
+# https://stackoverflow.com/questions/697560/how-to-copy-directory-from-source-tree-to-binary-tree
+macro(CONFIGURE_FILES srcDir destDir)
+  make_directory(${destDir})
+  file(GLOB_RECURSE templateFiles RELATIVE ${srcDir} ${srcDir}/*)
+
+  foreach(templateFile ${templateFiles})
+    set(srcTemplatePath ${srcDir}/${templateFile})
+    message(${srcTemplatePath})
+
+    if(NOT IS_DIRECTORY ${srcTemplatePath})
+      configure_file(${srcTemplatePath}
+                     ${destDir}/${templateFile}
+                     @ONLY)
+    endif()
+
+  endforeach(templateFile)
+
+endmacro(CONFIGURE_FILES)
+
+# Allow specifying the robot name
+set(ROBOT_NAME "" CACHE STRING "Select the the robot name that will match YARP_ROBOT_NAME")
```

Comment:
Should this refer to [3] instead of [1]  https://github.com/robotology-playground/icub-gazebo-wholebody/pull/12#issuecomment-318604563  ?

Reply:
All these names still confuse me. Would it be better the following description of the option?
```cmake
set(ROBOT_NAME "" CACHE STRING "Select the the robot name that will also match the sdf's model://{ROBOT_NAME} (model folder searched by Gazebo)")
```

Reply:
Great!